src/Groups/matrices/matrix_manipulation.jl

```diff
@@ -238,6 +238,32 @@ function ishermitian_matrix(B::MatElem{T}) where T <: FinFieldElem
    return true
 end
 
+# return (true, h) if y = hx, (false, nothing) otherwise
+function _is_scalar_multiple_mat(x::MatElem{T}, y::MatElem{T}) where T <: FieldElem
+   F=base_ring(x)
+   F==base_ring(y) || return (false, nothing)
+   nrows(x)==nrows(y) || return (false, nothing)
+   ncols(x)==ncols(y) || return (false, nothing)
+
+   if x==0
+      if y==0 return (true, F(1))
+      else return (false, nothing)
+      end
+   end
+
+   h = F(1)
+   for i in 1:nrows(x), j in 1:ncols(x)
+      if !iszero(x[i,j])
+         h = y[i,j] * x[i,j]^-1
+         break
+      end
+   end
+
+   y == h*x || return (false, nothing)
+   return (true,h)
+
+end
```

Comment:
Following what @fieker and me said, you can simplify this:
```suggestion
   for i in 1:nrows(x), j in 1:ncols(x)
      if !iszero(x[i,j])
         h = y[i,j] / x[i,j]
         return y == h*x ? (true,h) : (false, nothing)
      end
   end
  
   # at this point, x must be zero
   return y == 0 ? (true, F(1)) : (false, nothing)
end
```

Reply:
Corrected.
Just replaced `y[i,j]/x[i,j]` by y[i,j]*x[i,j]^-1`, since the operator `/` does not work on elements of type `fq_nmod`.